refactor(inkless): reuse storage metrics instance on cache

jeqo opened the pull request:
Instead of creating yet another metrics instance, reuse the storage metrics instance used by storage layer, as caching is a related module.

This will change the metric name prefix from:
`io.aiven.inkless.cache.caffeine` to `io.aiven.inkless.storage`

Related fixes:
- Add missing headers to cache classes
- Add tests to ensure metric names do not change without notice

jeqo committed:
refactor(inkless): reuse storage metrics instance on cache

Instead of creating yet another metrics instance, reuse the storage
metrics instance used by storage layer, as caching is a related module.

This will change the metric name prefix from:
`io.aiven.inkless.cache.caffeine` to `io.aiven.inkless.storage`

jjaakola-aiven reviewed (dismissed):
Naming of the metric is now unclear in the PR. It would be good to have a test to validate that metrics names are expected and not changed by mistake in future as the change will propagate to ops dashboards everywhere.

For example the cache size metrics name would be `io.aiven.inkless.storage.wal-segment-cache.size`, right?

jeqo committed:
fixup! refactor(inkless): reuse storage metrics instance on cache

Adding missing headers

jeqo committed:
fixup! refactor(inkless): reuse storage metrics instance on cache

On a second thought, we could leave it as is as it may slow down shuting
down the broker. Also, a bit outside the scope of the PR. Let's revisit if needed.

jeqo committed:
fixup! refactor(inkless): reuse storage metrics instance on cache

Add tests
